vm/src/error.rs

```diff
@@ -27,10 +27,10 @@ impl ActorError {
         }
     }
 
-    pub fn new_fatal(msg: String) -> Self {
+    pub fn new_fatal(exit_code: ExitCode, msg: String) -> Self {
```

Comment:
```suggestion
    pub fn new_fatal(msg: String) -> Self {
```
Exit code should never matter for a fatal error, and there would be nothing to match against the go client here
(I think we will move to this exit code being optional very soon)